### PR TITLE
Wp-admin: Add notice for Search customers without the constant

### PR DESCRIPTION
**Pull request by rebeccahum**
## Description
Since March 1 will be here before we know it, it would be good to display a notice.
Screenshot 2023-01-25 at 3 11 28 PM

### Updated: Wp-admin

Add reminder to test out Enterprise Search upgrade 

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test
1) Define search constant:
```
define( 'VIP_ENABLE_VIP_SEARCH', true ); // Enables Enterprise Search
```
2) Hit up wp-admin with super admin
3) Expect to see notice
4) Add testing constant:
```
define( 'VIP_SEARCH_USE_NEXT_EP', true );
```
5) Repeat step 2 
6) Expect to see notice gone

### CODE REVIEW

**Comment by GaryJones** [comment]
Since admin notices can come from different sources (plugins), then perhaps make it clearer that this is related to WPVIP / the platform, so folks don't just dismiss it as unimportant?

Also, who is this targeted at? Seems like everyone? Which means a lot of authors, editors and other roles that can't take any action, or won't know what Enterprise Search is, will also be seeing it (as noise).

**Comment by rebeccahum** [comment]
@GaryJones This only displays it for Enterprise Search customers and users with a super admin. However, I've amended the messaging to indicate that this is on WordPress VIP.

**Comment by rinatkhaziev** [comment]
> Since admin notices can come from different sources (plugins), then perhaps make it clearer that this is related to WPVIP / the platform, so folks don't just dismiss it as unimportant?

One would hope that that someone who has explicitly Enterprise Search constant set in the code is well-aware what Enterprise Search is enabled for their site?

**Comment by GaryJones** [comment]
If they are a developer, sure, but not all site/super-admins are developers.

File: admin-notice/admin-notice.php [hunk 1]
```diff
@@ -47,3 +47,23 @@ function( $admin_notice_controller ) {
 		);
 	}
 );
+
+// Enterprise Search upgrade
+add_action(
+	'vip_admin_notice_init',
+	function( $admin_notice_controller ) {
+		global $wp_version;
```
Comment (GaryJones) [inline admin-notice/admin-notice.php]: Why is this line needed?

Comment (rebeccahum) [reply admin-notice/admin-notice.php]: Oops. Bad copy pasta.

**Review by rinatkhaziev — APPROVED** [review]
